src/selective.d

```diff
@@ -277,16 +277,45 @@ private bool isPathExcluded(string path, string[] allowedPaths)
 		
 		// Is path is an exact match of the allowed path?
 		if (comm.length == path.length) {
-			// the given path is contained in an allowed path
-			if (!exclude) {
-				log.vdebug("Evaluation against 'sync_list' result: direct match");
-				finalResult = false;
-				// direct match, break and go sync
-				break;
+			// we have a potential exact match
+			// strip any potential '/*' from the allowed path, to avoid a potential lesser common match
+			string strippedAllowedPath = strip(allowedPath[offset..$], "/*");
+			
+			if (path == strippedAllowedPath) {
+				// we have an exact path match
+				log.vdebug("exact path match");
+				if (!exclude) {
+					log.vdebug("Evaluation against 'sync_list' result: direct match");
+					finalResult = false;
+					// direct match, break and go sync
+					break;
+				} else {
+					log.vdebug("Evaluation against 'sync_list' result: direct match but to be excluded");
+					finalResult = true;
+					// do not set excludeMatched = true here, otherwise parental path also gets excluded
+				}	
 			} else {
-				log.vdebug("Evaluation against 'sync_list' result: direct match but to be excluded");
-				finalResult = true;
-				// do not set excludeMatched = true here, otherwise parental path also gets excluded
+				// no exact path match, but something common does match
+				log.vdebug("something 'common' matches the input path");
+				auto splitAllowedPaths = pathSplitter(strippedAllowedPath);
+				string pathToEvaluate = "";
+				foreach(base; splitAllowedPaths) {
+					pathToEvaluate ~= base;
+					if (path == pathToEvaluate) {
+						// The input path matches what we want to evaluate against as a direct match
+						if (!exclude) {
+							log.vdebug("Evaluation against 'sync_list' result: direct match for parental path item");
+							finalResult = false;
+							// direct match, break and go sync
+							break;
```

Comment:
This `break` will stop only the new foreach of line 302 but not the one on line 232, is that expected?  
I don't think it changes anything functionally (as we should put excludes before includes in sync_list), but could be better/optimized to break from first foreach (L232) to avoid testing against all the remaining allowedPaths

Reply:
As per the comments on lines 230/231:
```
	// unless path is an exact match, entire sync_list entries need to be processed to ensure
	// negative matches are also correctly detected
```

So no .. the break when checking a parental path is correct, and not breaking out of checking all the lines is also correct for the current logic.